test/system_SUITE.erl

```diff
@@ -0,0 +1,461 @@
+                                                               {key, KeyFirst} ]}),
+    %% verify the other certificate is not accepted
+    {error, ?SERVER_REJECT_CLIENT} =
+     amqp_connection:start(#amqp_params_network{host = Host,
+                                                port = Port,
+                                                ssl_options = [{cert, CertUpdated},
+                                                               {key, KeyUpdated} ]}),
+    ok = amqp_connection:close(Con),
+
+    %% When: a whitelisted certicate is replaced with one with the same name
+    ok = whitelist(Config, "bart", CertUpdated,  KeyUpdated),
+
+    wait_for_trust_store_refresh(),
+
+    %% Then: the first certificate should be rejected
+    {error, ?SERVER_REJECT_CLIENT} =
+     amqp_connection:start(#amqp_params_network{host = Host,
+                                                port = Port,
+                                                ssl_options = [{cert, CertFirst},
+                                                               %% disable ssl session caching
+                                                               %% as this ensures the cert
+                                                               %% will be re-verified by the
+                                                               %% server
+                                                               {reuse_sessions, false},
+                                                               {key, KeyFirst} ]}),
+
+    %% And: the updated certificate should allow the user to connect
+    {ok, Con2} =
+     amqp_connection:start(#amqp_params_network{host = Host,
+                                                port = Port,
+                                                ssl_options = [{cert, CertUpdated},
+                                                               {reuse_sessions, false},
+                                                               {key, KeyUpdated} ]}),
+    ok = amqp_connection:close(Con2),
+    %% Clean: server TLS/TCP.
+    ok = delete("bart.pem", Config),
+    ok = rabbit_networking:stop_tcp_listener(Port).
+
+
+ensure_configuration_using_binary_strings_is_handled(Config) ->
+    ok = rabbit_ct_broker_helpers:rpc(Config, 0,
+           ?MODULE, ensure_configuration_using_binary_strings_is_handled1, [Config]).
+
+ensure_configuration_using_binary_strings_is_handled1(Config) ->
+    ok = change_configuration(rabbitmq_trust_store, [{directory, list_to_binary(friendlies(Config))},
+                                                    {refresh_interval,
+                                                        {seconds, interval()}}]).
+
+%% Test Constants
+
+port(Config) ->
+    rabbit_ct_broker_helpers:get_node_config(Config, 0, tcp_port_amqp_tls_extra).
+
+data_directory(Config) ->
+    Path = ?config(priv_dir, Config),
+    % error_logger:info_msg("config: ~p", [Config]),
+    %% Ensure TMPDIR exists
+    true = (false =/= Path),
```

Comment:
This variable is provided by common_test so there is no need to test it. You can use `?config(priv_dir, Config)` directly in the function below.